internal/resolver/resolver.go

```diff
@@ -57,6 +57,11 @@ func (r *Resolver) Resolve(reqs []*chart.Dependency, repoNames map[string]string
 	locked := make([]*chart.Dependency, len(reqs))
 	missing := []string{}
 	for i, d := range reqs {
+		constraint, err := semver.NewConstraint(d.Version)
+		if err != nil {
+			return nil, errors.Wrapf(err, "dependency %q has an invalid version/constraint format", d.Name)
+		}
+
 		if d.Repository == "" {
 			// Local chart subfolder
```

Comment:
I just realized that this doesn't fix #9348. In #9348 the repository exists as it starts with `file://`. The modified code path here is for the case where there is no repository set. Instead, this code touches on the code path for charts who exist only in the `charts/` sub-directory of a chart. There is no outside location as no repository is set.

Can this PR be updated to fix #9348, please.

Reply:
Sorry, I didn't quite get it, the modified code path here is for the case where that the repository exists.

See, line: 73

Reply:
Actually, it was my fault. It does resolve the issue. I was reading it wrong in the GitHub interface. Some code was collapsed. It's good.